www/caddy/src/opnsense/mvc/app/models/Pischem/Caddy/Migrations/M1_1_3.php

```diff
@@ -0,0 +1,87 @@
+ *    AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ *    OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *    SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *    INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *    CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *    ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *    POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Pischem\Caddy\Migrations;
+
+use OPNsense\Base\BaseModelMigration;
+use OPNsense\Core\Config;
+
+class M1_1_3 extends BaseModelMigration
+{
+    public function run($model)
+    {
+        // Load the current system configuration
+        $config = Config::getInstance()->object();
+        
+        // Track whether any changes are made during the migration
+        $hasChanges = false;
+
+        // Ensure there are reverse proxy configurations to process
+        if (!empty($config->Pischem->caddy->reverseproxy)) {
+        
+            // Loop through each reverse proxy configuration in the stored configuration config.xml
+            foreach ($config->Pischem->caddy->reverseproxy->children() as $configNode) {
+            
+                // Extract the UUID attribute to identify the configuration item
+                $uuid = (string)$configNode->attributes()->uuid;
+                
+                // Check if the current configuration item has a 'Description' to migrate
+                if (!empty($configNode->Description)) {
+                
+                    // Store the value of 'Description' for migration.
+                    $descriptionValue = (string)$configNode->Description;
+
+                    // Attempt to locate the corresponding node in the model using the UUID
+                    $modelNode = null;
+                    
+                    // Retrieve reverse proxy items from the model for matching UUID
+                    $reverseProxies = $model->getNodeByReference('reverseproxy')->iterateItems();
+                    foreach ($reverseProxies as $item) {
+                        foreach ($item->iterateItems() as $modelUuid => $node) {
+                            if ($uuid === $modelUuid) {
+                                $modelNode = $node;
+                                break 2; // Break from both loops once the node is found
+                            }
+                        }
+                    }
+
+                    // If a matching node is found in the model, migrate the 'Description' value to 'description' value
+                    if ($modelNode !== null) {
+                        $modelNode->description = $descriptionValue;
+                        $hasChanges = true;
+                    }
+                }
+            }
+        }
+
+        // Save model if changes have been made
+        if ($hasChanges) {
+            parent::run($model);
```

Comment:
$has_changes is wrong here, the parent::run() is for filling (missing) default values, which you don't even have in the migration. you can delete all of its use and the conditional.

Reply:
Thanks for explaining that, I have removed these. Still working as expected for me in my tests. f3db69d

Reply:
👍